client/pluginmanager/csimanager/volume.go

```diff
@@ -301,19 +316,32 @@ func (v *volumeManager) unpublishVolume(ctx context.Context, vol *structs.CSIVol
 	return rpcErr
 }
 
-func (v *volumeManager) UnmountVolume(ctx context.Context, vol *structs.CSIVolume, alloc *structs.Allocation, usage *UsageOptions) error {
+func (v *volumeManager) UnmountVolume(ctx context.Context, vol *structs.CSIVolume, alloc *structs.Allocation, usage *UsageOptions) (err error) {
 	logger := v.logger.With("volume_id", vol.ID, "alloc_id", alloc.ID)
 	ctx = hclog.WithContext(ctx, logger)
 
-	err := v.unpublishVolume(ctx, vol, alloc, usage)
-	if err != nil {
-		return err
+	err = v.unpublishVolume(ctx, vol, alloc, usage)
+
+	if err == nil {
+		canRelease := v.usageTracker.Free(alloc, vol, usage)
+		if v.requiresStaging && canRelease {
+			err = v.unstageVolume(ctx, vol, usage)
+		}
 	}
 
-	canRelease := v.usageTracker.Free(alloc, vol, usage)
-	if !v.requiresStaging || !canRelease {
-		return nil
+	event := structs.NewNodeEvent().
+		SetSubsystem(structs.NodeEventSubsystemStorage).
+		SetMessage("Unmount volume").
+		AddDetail("Volume namespace", vol.Namespace).
+		AddDetail("Volume ID", vol.ID)
+	if err == nil {
+		event.AddDetail("Success", "true")
```

Comment:
Maybe nitpicky, but everywhere else we create node events, the subsystem is `PascalCased`, the message is `Sentence cased`, and the detail fields are `snake_cased`.

Reply:
Good call, changed